ddtrace/contrib/pytest/plugin.py

```diff
@@ -84,13 +116,84 @@ def pytest_configure(config):
     config.addinivalue_line("markers", "dd_tags(**kwargs): add tags to current span")
     if is_enabled(config):
         _CIVisibility.enable(config=ddtrace.config.pytest)
+        test_command_span = _CIVisibility._instance.tracer.trace(
+            "pytest.test_session",
+            service=_CIVisibility._instance._service,
+            span_type=SpanTypes.TEST,
+        )
+        test_command_span.set_tag_str(COMPONENT, "pytest")
+        test_command_span.set_tag_str(SPAN_KIND, KIND)
+        test_command_span.set_tag_str(test.FRAMEWORK, FRAMEWORK)
+        test_command_span.set_tag_str(test.FRAMEWORK_VERSION, pytest.__version__)
+        test_command_span.set_tag_str(_EVENT_TYPE, "test_session_end")
+        test_command_span.set_tag_str(test.COMMAND, _get_pytest_command(config))
+        test_command_span.set_tag(_SESSION_ID, test_command_span.trace_id)
 
 
 def pytest_sessionfinish(session, exitstatus):
+    """
+    Hooks right after a whole test session has finished, right before returning the exit status to system.
+    """
     if is_enabled(session.config):
+        test_command_span = _CIVisibility._instance.tracer.current_root_span()
+        if test_command_span is not None:
+            test_command_span.finish()
         _CIVisibility.disable()
 
 
+@pytest.fixture(scope="module", autouse=True)
+def ddtrace_test_module_fixture(request):
+    """
+    Hook to trace test modules in pytest. This fixture runs during the setup/teardown of every pytest test module,
+    which is as close as possible to the start/end of a test suite. Pytest doesn't offer hooks at the start of a test
+    suite, so we have to use a module-scoped fixture instead.
+    """
+    if not _CIVisibility.enabled:
+        yield
+        return
+    with _CIVisibility._instance.tracer.trace(
+        "pytest.test_module",
+        service=_CIVisibility._instance._service,
+        span_type=SpanTypes.TEST,
+    ) as test_module_span:
+        test_module_span.set_tag_str(COMPONENT, "pytest")
+        test_module_span.set_tag_str(SPAN_KIND, KIND)
+        test_module_span.set_tag_str(test.FRAMEWORK, FRAMEWORK)
+        test_module_span.set_tag_str(test.FRAMEWORK_VERSION, pytest.__version__)
+        test_module_span.set_tag_str(_EVENT_TYPE, "test_suite_end")
+        test_module_span.set_tag(_SESSION_ID, test_module_span.trace_id)
+        test_module_span.set_tag(_SUITE_ID, test_module_span.span_id)
```

Comment:
test suites in CI Visibility should also include `_MODULE_ID`

Reply:
It's possible (thought not usual) in pytest to run a specific test file (suite) only not contained in a test package, meaning sometimes a package might not exist. I've added this, but also with a check to account for this edge case as well.